src/Test.UnitTests.BinSkim.Driver/MultithreadedAnalyzeCommandTests.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Text;
+
+using FluentAssertions;
+
+using Microsoft.CodeAnalysis.IL;
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.BinSkim.Rules
+{
+    public class MultithreadedAnalyzeCommandTests
+    {
+        [Fact]
+        public void MultithreadedAnalyzeCommand_ReturnCommonPathRootFromTargetSpecifiersIfOneExists()
+        {
+            var testCases = new[]
+            {
+                new
+                {
+                    TargetFileSpecifiers = new[]
+                    {
+                        @"C:\*.dll",
+                        @"D:\*.dll"
+                    },
+                    ExpectedCommonPath = string.Empty
+                },
+                new
+                {
+                    TargetFileSpecifiers = new[]
+                    {
+                        @"C:\path1\",
+                        @"C:\path2\"
+                    },
+                    ExpectedCommonPath = string.Empty
+                },
+                new
+                {
+                    TargetFileSpecifiers = new[]
+                    {
+                        @"C:\path1\",
+                        @"C:\path1\path2\"
+                    },
+                    ExpectedCommonPath = string.Empty
+                },
+                new
+                {
+                    TargetFileSpecifiers = new[]
+                    {
+                        @"C:\path1\*.dll",
+                        @"C:\path1\*.exe"
```

Comment:
>path1 [](http://example.com/codeflow?start=29&length=5)

i don't see mixes of files and directories. I'd also mix full names + wildcards.